core/tools.js

```diff
@@ -1991,6 +1991,59 @@
 				}
 
 				return null;
+			},
+
+			/**
+			 * Merges two objects and returns new one.
+			 *
+			 *		var obj1 = {
+			 *				a: 1,
+			 *				conflicted: 10,
+			 *				obj: {
+			 *					c: 1
+			 *				}
+			 *			},
+			 *			obj2 = {
+			 *				b: 2,
+			 *				conflicted: 20,
+			 *				obj: {
+			 *					d: 2
+			 *				}
+			 *			};
+			 *
+			 *		CKEDITOR.tools.object.merge( obj1, obj2 );
+			 *
+			 * This code produces the following object;
+			 *
+			 *		{
+			 *			a: 1,
+			 *			b: 2,
+			 *			conflicted: 20,
+			 *			obj: {
+			 *				c: 1,
+			 *				d: 2
+			 *			}
+			 *		}
+			 *
+			 * @param {Object} obj1 A base object.
+			 * @param {Object} obj2 An object, which properties will be merged to the base one.
```

Comment:
> An object, which properties will be merged to the base one.

I think this is a little misleading, as properties are not merged to the base object, but to its copy, so the base object is not modified. And it is also inconsistent with method description which suggest the new object is created (_Merges two objects and returns new one_).

ATM I don't have any clear idea how it could be described. I checked in other libs/docs how it is described there:

* https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign
* https://lodash.com/docs/#merge
* http://api.jquery.com/jQuery.merge/
* http://ramdajs.com/docs/#merge

but still not sure. Maybe you could come up with something reasonable?

Reply:
Maybe something like that?

```
obj1 Source object, which will be used to create a new base object.
```

And then `obj2` description is already good :D WDYT @f1ames?

Reply:
@Comandeer works for me 👍